### PR TITLE
#28: Adding multi-threading to scraping process for massive speed improvements

### PR DESCRIPTION
**Pull request by dmhacker**

### CODE REVIEW

File: backend/database/.gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-*
```
Comment (CTrando) [inline backend/database/.gitignore]: why did you delete this?

File: backend/scraper/scraper.py [hunk 2]
```diff
@@ -29,83 +29,114 @@ def __init__(self):
         self.login_url = SCHEDULE_OF_CLASSES
 
         # Connecting to the database for the list of departments
+        os.makedirs(DATABASE_FOLDER_PATH, exist_ok=True)
```
Comment (CTrando) [inline backend/scraper/scraper.py]: I like this

File: backend/settings.py [hunk 3]
```diff
@@ -12,12 +12,17 @@
...
+# DATABASE_PATH = os.path.join(HOME_DIR, 'database', 'data.db')
+DATABASE_FOLDER_PATH = "/cache/database"
+DATABASE_PATH = os.path.join(DATABASE_FOLDER_PATH, "data.db")
+
 # Image directory
 IMAGE_DIR = os.path.join(HOME_DIR, 'images')
 
 # Where the classes are stored
-HTML_STORAGE = os.path.join(HOME_DIR, 'courseNums')
+# HTML_STORAGE = os.path.join(HOME_DIR, 'courseNums')
```
Comment (CTrando) [inline backend/settings.py]: don't leave comments in there

File: backend/scraper/scraper.py (hunk 2 above)
Comment (dmhacker) [reply backend/scraper/scraper.py]: good

File: backend/database/.gitignore (hunk 1 above)
Comment (dmhacker) [reply backend/database/.gitignore]: It's no longer necessary. Whenever we perform scraping, we create a /cache/database folder in the Docker container that will hold the SQLite scratchpad database. This folder is basically useless.

File: backend/settings.py (hunk 3 above)
Comment (dmhacker) [reply backend/settings.py]: I will fix this.
